esphome/components/http_request/http_request.cpp

```diff
@@ -38,18 +40,21 @@ void HttpRequestComponent::send(const std::vector<HttpRequestResponseTrigger *>
 
   bool begin_status = false;
   const String url = this->url_.c_str();
-#ifdef USE_ESP32
-  begin_status = this->client_.begin(url);
-#endif
-#ifdef USE_ESP8266
-#if ARDUINO_VERSION_CODE >= VERSION_CODE(2, 7, 0)
-  this->client_.setFollowRedirects(HTTPC_STRICT_FOLLOW_REDIRECTS);
-#elif ARDUINO_VERSION_CODE >= VERSION_CODE(2, 6, 0)
-  this->client_.setFollowRedirects(true);
+#if defined USE_ESP32 || defined ESP8266 && ARDUINO_VERSION_CODE >= VERSION_CODE(2, 6, 0)
+#if defined USE_ESP32 || ARDUINO_VERSION_CODE >= VERSION_CODE(2, 7, 0)
```

Comment:
Let's add some braces to make it a bit more clear that we only check the version on ESP8266:
```suggestion
#if defined(USE_ESP32) || (defined(ESP8266) && ARDUINO_VERSION_CODE >= VERSION_CODE(2, 6, 0))
#if defined(USE_ESP32) || ARDUINO_VERSION_CODE >= VERSION_CODE(2, 7, 0)
```